main.tf

```diff
@@ -549,10 +549,11 @@ resource "aws_cloudfront_distribution" "default" {
         }
       }
 
-      viewer_protocol_policy = ordered_cache_behavior.value.viewer_protocol_policy
-      default_ttl            = ordered_cache_behavior.value.default_ttl
-      min_ttl                = ordered_cache_behavior.value.min_ttl
-      max_ttl                = ordered_cache_behavior.value.max_ttl
+      viewer_protocol_policy     = ordered_cache_behavior.value.viewer_protocol_policy
+      default_ttl                = ordered_cache_behavior.value.default_ttl
+      min_ttl                    = ordered_cache_behavior.value.min_ttl
+      max_ttl                    = ordered_cache_behavior.value.max_ttl
+      response_headers_policy_id = try(ordered_cache_behavior.value.response_headers_policy_id, null)
```

Comment:
You don't need the `try` here. Because `var. ordered_cache` is an object, all of the attributes specified in the type need to be passed. So this will always have a value of at least `null` or `""` from the module consumer. Mind removing as it may confuse other contributors in the future?

Reply:
I would like to keep the try function because there is no optional function in Terraform variables, which means that for custom behavior you have to specify this parameter, which uselessly increases the amount of code
https://github.com/hashicorp/terraform/issues/19898

Reply:
The `ordered_cache` variable is also declared without default values: https://github.com/cloudposse/terraform-aws-cloudfront-s3-cdn/blob/master/variables.tf#L412

Reply:
@eboboshka Yeah, so for both of those points that means that object's value will have to provide a value for the attribute when it is passed from the root module, regardless of the `try` or not. `tf validate` (and the various commands that run `validate` as a pre-step) will not allow a consumer to pass a value to `ordered_cache` without it including `response_headers_policy_id`. So the `try` becomes redundant as it will always be given a value since there is no option to do otherwise.

Reply:
You're probably right. I'm just trying to remove unnecessary options since I'm using your module as a for_each module.
Anyway, if you think it''s necessary to remove `try`, you can change my commit

Just as an example
![image](https://user-images.githubusercontent.com/31067306/150975829-49b85b16-c4c8-4993-9535-be07728e7a1c.png)
![image](https://user-images.githubusercontent.com/31067306/150975977-610ccfe7-e1f9-41be-91d4-6468a4fb22d8.png)

Reply:
Thanks for the update @eboboshka!